applications/vanilla/controllers/class.discussioncontroller.php

```diff
@@ -82,7 +82,9 @@ public function index($DiscussionID = '', $DiscussionStub = '', $Page = '') {
         // Check permissions.
         $Category = CategoryModel::categories($this->Discussion->CategoryID);
         $this->permission('Vanilla.Discussions.View', true, 'Category', val('PermissionCategoryID', $Category, -1));
-        $this->setData('CategoryID', $this->CategoryID = $this->Discussion->CategoryID, true);
+        if (c('Vanilla.Categories.Use', true)) {
+            $this->setData('CategoryID', $this->CategoryID = $this->Discussion->CategoryID, true);
```

Comment:
We're going from a scenario where CategoryID in the controller's `$Data` array is always set to having it only set some of the time.  Since the controller's data array is public, there's a chance something might try to access a potentially-invalid index.

What about something like this?

```php
$this->CategoryID = c('Vanilla.Categories.Use', true) ? $this->Discussion->CategoryID : null;
$this->setData('CategoryID', $this->CategoryID, true);
```